### ISSUE TITLE
[TypeScript plugin] Supporting enums generated with integer values instead of strings

### ISSUE THREAD

**Issue by theomjones** [dotansimha/graphql-code-generator#4157]
**Is your feature request related to a problem? Please describe.**

Generating from schema enums uses the keys as values when it should use integers.

Given the following enum in a schema:
```graphql
enum Status {
  RUNNING
  IDLE
  ERROR
}
```
The generated typescript enum I would expect is
```ts
enum Status {
  RUNNING, // 0
  IDLE, // 1
  ERROR // 2
}
```
However the actual output is:

```ts
enum Status {
  RUNNING = "RUNNING",
  IDLE = "IDLE",
  ERROR = "ERROR"
}
```

**Describe the solution you'd like**

- Adding a configuration option to not transform enum values to their keys could be a solution

**Describe alternatives you've considered**

- I have tried all the enum configuration options currently supported in the typescript plugin but can't find one that helps with this.

**Additional context**

 This makes it difficult to do frontend conditional checks against enum values. E.g. I'm checking that the returned `Status` from a query is of a certain enum value. In my case the one on the graphql response will be an integer, however the generated enum value is now a string – so checking equality is not possible.

**Comment by killjoy2013** [comment]
@dotansimha @ardatan I'd like to work on this. Please assign to me if you consider appropriate.

**Comment by dotansimha** [comment]
Hi @killjoy2013, thank you!
@theomjones is already working on it, progress is here: https://github.com/dotansimha/graphql-code-generator/pull/4158

**Comment by dotansimha** [comment]
@theomjones we'll release a new version very soon. You can use `1.15.2-alpha-a94cba0a.0` in the meantime.

**Comment by dotansimha** [comment]
Fixed in `v1.15.2`
